src/main/java/io/cdap/plugin/servicenow/sink/ServiceNowSinkConfig.java

```diff
@@ -175,21 +135,18 @@ public void validateSchema(Schema schema, FailureCollector collector) {
       || containsMacro(ServiceNowConstants.PROPERTY_OPERATION) || containsMacro(PROPERTY_EXTERNAL_ID_FIELD)) {
       return;
     }
-
-    Schema tableSchema = getTableMetaData(tableName, this).getSchema();
+    ServiceNowTableAPIClientImpl restApi = new ServiceNowTableAPIClientImpl(this);
+    Schema tableSchema = restApi.fetchServiceNowTableSchema(tableName, collector);
```

Comment:
need to check if tableSchema is null here

Reply:
Added null check. f453a61bee4a0cffcfd69509adb54fa1b7933f27